### PR TITLE
NP-262 Upgrading to Scala 3.3.3

### PR DESCRIPTION
**Pull request by AlessandroCinque**
https://jira.tools.tax.service.gov.uk/browse/NP-262

### CODE REVIEW

**Comment by AlessandroCinque** [comment]
image
...
image

File: build.sbt [hunk 1]
```diff
@@ -2,8 +2,9 @@ lazy val root = (project in file("."))
   .settings(
     name := "uknw-auth-checker-api-tests",
     version := "0.1.0",
-    scalaVersion := "2.13.12",
-    libraryDependencies ++= Dependencies.test
+    scalaVersion := "3.3.3",
+    libraryDependencies ++= Dependencies.test,
+    (Compile / compile) := ((Compile / compile) dependsOn (Compile / scalafmtSbtCheck, Compile / scalafmtCheckAll)).value
```
Comment (TheDanielDoyle) [inline build.sbt:7]: This looks like a merge conflict. The compile/compile line was removed in a previous PR.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
-
+  val PekkoVersion        = "1.1.0-M1"
   val test: Seq[ModuleID] = Seq(
-    "com.typesafe"         % "config"                  % "1.4.3"  % Test,
-    "com.typesafe.play"   %% "play-ahc-ws-standalone"  % "2.2.7"  % Test,
-    "com.typesafe.play"   %% "play-ws-standalone-json" % "2.2.7"  % Test,
-    "com.vladsch.flexmark" % "flexmark-all"            % "0.64.8" % Test,
-    "org.scalatest"       %% "scalatest"               % "3.2.18" % Test,
-    "org.slf4j"            % "slf4j-simple"            % "2.0.13" % Test
+    "org.apache.pekko"    %% "pekko-actor-typed"         % PekkoVersion,
```
Comment (TheDanielDoyle) [inline project/Dependencies.scala:6]: You shouldn't need to bring in Pekko directly.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+    "org.apache.pekko"    %% "pekko-stream"              % PekkoVersion,
+    "org.apache.pekko"    %% "pekko-slf4j"               % PekkoVersion,
+    "org.apache.pekko"    %% "pekko-actor-testkit-typed" % PekkoVersion  % Test,
+    "org.commonmark"       % "commonmark"                % "0.22.0"      % Test,
+    "com.typesafe"         % "config"                    % "1.4.3"       % Test,
+    "org.playframework"   %% "play-ahc-ws-standalone"    % "3.1.0-M2"    % Test,
```
Comment (TheDanielDoyle) [inline project/Dependencies.scala:12]: We should be using the latest stable version. In this case it's 3.0.5. Same for the one below.

File: src/test/scala/uk/gov/hmrc/api/client/HttpClient.scala [hunk 4]
```diff
@@ -16,26 +16,34 @@
...
 import play.api.libs.ws.DefaultBodyWritables._
 import play.api.libs.ws.JsonBodyWritables.writeableOf_JsValue
-import play.api.libs.ws.StandaloneWSRequest
+import play.api.libs.ws.{StandaloneWSRequest, StandaloneWSResponse}
 import play.api.libs.ws.ahc.StandaloneAhcWSClient
 import uk.gov.hmrc.api.conf.ZapConfiguration._
 
 import scala.concurrent.{ExecutionContext, Future}
 
+object MyActor {
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/api/client/HttpClient.scala:30]: Why is this necessary?

File: src/test/scala/uk/gov/hmrc/api/client/HttpClient.scala [hunk 5]
```diff
@@ -16,26 +16,34 @@
...
+    context.log.info(s"Received message: $message")
+    Behaviors.same
+  }
+}
+
 trait HttpClient {
 
-  implicit val actorSystem: ActorSystem = ActorSystem()
-  implicit val ec: ExecutionContext     = ExecutionContext.global
+  implicit val actorSystem: ActorSystem[String] = ActorSystem(MyActor(), "myActorSystem")
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/api/client/HttpClient.scala:39]: implicit lazy val actorSystem: ActorSystem        = ActorSystem()

This should be enough.

**Review by TheDanielDoyle — APPROVED** [review]
